searchcorespi/src/vespa/searchcorespi/index/index_disk_dir_active_state.h

```diff
@@ -0,0 +1,22 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#pragma once
+
+namespace searchcorespi::index {
+
+/*
+ * Class describing active state for a disk index directory.
+ */
+class IndexDiskDirActiveState {
+    uint32_t _active_count;
+public:
+    IndexDiskDirActiveState()
+        : _active_count(0)
+    {
+    }
+
+    void activate() noexcept { ++_active_count; }
+    void deactivate() noexcept { --_active_count; }
```

Comment:
Should there be sanity checking to avoid going past 0.
Or should activate return a guard that did deactivate in the destructor ?

Reply:
Sanity check added. Guard not added.